### PR TITLE
feature/boilerplate

### PR DESCRIPTION
**Pull request by bradlocking**
Readme incoming once approved and merged

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  "parserOptions": {
+    "ecmaVersion": 6,
+    "ecmaFeatures": {
+      "jsx": true
+    }
+  },
+  "extends": [
+    "eslint:recommended",
+    "airbnb",
+    "ostmodern"
```
Comment (inestavares) [inline .eslintrc]: We might not want to give them our linter...

Comment (inestavares) [reply .eslintrc]: I would actually not give them any linter. I'd like to see how they write code

File: app/index.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+@import 'styles/fonts';
+
+.example-class {
+  font-size: 50px;
+  font-family: 'FSAlbert';
```
Comment (inestavares) [inline app/index.scss]: Maybe not give them our fonts either

File: app/index.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import ReactDOM from 'react-dom';
+import { Index } from 'index';
+
+jest.mock('react-dom', () => ({
+  render: jest.fn()
+}));
+
+describe('Ost Front-end Code Test', () => {
```
Comment (inestavares) [inline app/index.spec.jsx]: Nice one

File: jest.config.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    "!app/**/*.spec.js",
+    "!app/**/*.spec.jsx"
+  ],
+  "coverageDirectory": "test-reports/unit-test-reports/coverage",
+  "coverageReporters": [
+    "cobertura",
+    "lcov",
+    "text"
+  ],
+  "coverageThreshold": {
```
Comment (inestavares) [inline jest.config.json]: Are tests optional? If yes, then get rid of this

File: package.json [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    "react-test-renderer": "^16.0.0",
+    "regenerator-runtime": "^0.10.5",
+    "sass-loader": "^6.0.6",
+    "style-loader": "^0.18.2",
+    "uglifyjs-webpack-plugin": "^0.4.6",
+    "webpack": "^3.1.0",
+    "webpack-dev-server": "^2.5.1"
+  },
+  "dependencies": {
+    "axios": "^0.16.2",
```
Comment (inestavares) [inline package.json]: nope

File: package.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    "babel-polyfill": "^6.23.0",
+    "classnames": "^2.2.5",
+    "history": "^4.6.3",
+    "lodash": "^4.17.4",
```
Comment (inestavares) [inline package.json]: nope

File: package.json [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    "normalize.css": "^7.0.0",
+    "prop-types": "^15.5.10",
+    "react": "^16.0.0",
+    "react-addons-transition-group": "^15.6.2",
```
Comment (inestavares) [inline package.json]: nope

File: package.json [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    "react-dom": "^16.0.0",
+    "react-router": "^4.1.1",
+    "react-router-dom": "^4.1.2",
+    "react-transition-group": "^2.2.1",
```
Comment (inestavares) [inline package.json]: nope

File: package.json [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    "eslint-plugin-jest": "^20.0.3",
+    "eslint-plugin-jsx-a11y": "^5.0.0",
+    "eslint-plugin-react": "^7.1.0",
+    "extract-text-webpack-plugin": "^3.0.0",
+    "file-loader": "^0.11.2",
+    "html-webpack-plugin": "^2.29.0",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^20.0.4",
+    "jest-junit": "^3.1.0",
+    "node-sass": "^4.5.3",
```
Comment (inestavares) [inline package.json]: they are not forced to use sass...

File: package.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  "version": "1.0.0",
+  "description": "Ostmodern Front-end Code Test",
+  "main": "index.js",
+  "scripts": {
+    "test": "jest --config jest.config.json --watch",
+    "coverage": "jest --config jest.config.json --watch --coverage",
+    "serve": "webpack-dev-server --config webpack.config.js"
+  },
+  "author": "Ostmodern",
+  "license": "ISC",
```
Comment (inestavares) [inline package.json]: Don't put a license in, maybe

File: webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    }),
+    new ExtractTextPlugin({
+      filename: 'static/style.css'
+    })
+  ],
+  resolve: {
+    extensions: ['.js', '.jsx'],
+    modules: [
+      path.resolve(`${paths.base}`),
+      path.resolve(`${paths.base}/lib`),
```
Comment (inestavares) [inline webpack.config.js]: I wouldn't do this

File: .gitignore [hunk 12]
```diff
@@ -0,0 +1,41 @@
+# CI/tests
+cms/jshint.xml
+test_xml
+v2/tests/coverage
```
Comment (inestavares) [inline .gitignore]: take this out

File: app/index.html [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (inestavares) [inline app/index.html]: is this emmet :D

File: .eslintrc (hunk 1 above)
Comment (bradlocking) [reply .eslintrc]: YES! good idea

File: app/index.scss (hunk 2 above)
Comment (bradlocking) [reply app/index.scss]: Hm, I spoke with Max about whether we should supply fonts, shall we go for a default google font like Roboto?

File: jest.config.json (hunk 4 above)
Comment (bradlocking) [reply jest.config.json]: I don't think tests should be optional

File: package.json (hunk 5 above)
Comment (bradlocking) [reply package.json]: Nice catch, thanks

File: package.json (hunk 6 above)
Comment (bradlocking) [reply package.json]: 👍

File: app/index.scss (hunk 2 above)
Comment (inestavares) [reply app/index.scss]: Maybe... I am not sure about the licenses for the fonts we are using, hence my comment!

File: jest.config.json (hunk 4 above)
Comment (inestavares) [reply jest.config.json]: for CMS definitely not, but services might have different opinions

File: app/index.scss (hunk 2 above)
Comment (bradlocking) [reply app/index.scss]: Yes I agree with that

Comment (bradlocking) [reply app/index.scss]: Just gonna use helvetica

**Pull request merged**